### PR TITLE
[WIP] feat: vault instructions

### PR DESCRIPTION
**Pull request by b2kdaman**

### CODE REVIEW

**Comment by b2kdaman** [comment]
Will add tests later

File: src/programs/vault/VaultProgram.ts [hunk 1]
```diff
@@ -9,6 +9,20 @@ export enum VaultKey {
...
+  ActivateVault,
+  CombineVault,
+  RedeemShares,
+  WithdrawTokenFromSafetyDepositBox,
+  MintFractionalShares,
+  WithdrawSharesFromTreasury,
+  AddSharesToTreasury,
+  UpdateExternalPriceAccount,
+  SetVaultAuthority,
+}
```
Comment (b2kdaman) [inline src/programs/vault/VaultProgram.ts:24]: Decided to remove magic numbers from instructions code

**Review by kurpav — APPROVED** [review]

**Review by m-sebastiyan — APPROVED** [review]
Looks nice!

File: src/programs/vault/transactions/UpdateExternalPriceAccount.ts [hunk 2]
```diff
@@ -11,10 +11,7 @@ export class UpdateExternalPriceAccountArgs extends Borsh.Data<{
 }> {
   static readonly SCHEMA = new Map([
     ...ExternalPriceAccountData.SCHEMA,
-    ...this.struct([
-      ['instruction', 'u8'],
-      ['externalPriceAccount', ExternalPriceAccountData],
-    ]),
+    ...this.struct([['instruction', 'u8']]),
```
Comment (b2kdaman) [inline src/programs/vault/transactions/UpdateExternalPriceAccount.ts:14]: Didn't work this way at all.

Comment (kurpav) [reply src/programs/vault/transactions/UpdateExternalPriceAccount.ts:14]: Is it in sync with the latest metaplex foundation?

Comment (b2kdaman) [reply src/programs/vault/transactions/UpdateExternalPriceAccount.ts:14]: Yes it does.

**Review by kurpav — APPROVED** [review]

**Review by m-sebastiyan — APPROVED** [review]

**Pull request merged**